### DISCUSSION TITLE
addon

### DISCUSSION THREAD

**Discussion by starkodder** [ScratchAddons/ScratchAddons#2716] [Show and tell]
Hello I recently created an addon, [https://addons.mozilla.org/sv-SE/firefox/addon/studio-buttons/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search](https://addons.mozilla.org/sv-SE/firefox/addon/studio-buttons/?utm_source=addons.mozilla.org&utm_medium=referral&utm_content=search) do you want to add it?

**Comment by WorldLanguages** [comment]
Where's the source?

**Reply by starkodder** [reply]
https://github.com/starkodder/scratch-addon (you need to unzip it)

**Comment by Shluffy** [comment]
Would You please be able to put the source up on githu and then link it?

**Comment by starkodder** [comment]
here: `https://github.com/starkodder/scratch-addon`

**Reply by WorldLanguages** [reply]
It's a zip, can you upload file per file?

**Reply by GrahamSH-LLK** [reply]
The main code is
```js
var url = window.location.pathname.split("/");
setTimeout(() => { 
if(url[url.length-2] === "projects") {
	var element = document.getElementById("project-action-bar");
	var button = document.createElement('div');
	button.classList.add('button');
	button.classList.add('grey');
	button.classList.add('small');
	button.textContent = " Delete all projects ";
	button.addEventListener("click", function(){ for(var i = 0;i<document.getElementsByClassName("remove").length;i++) {
			document.getElementsByClassName("remove")[i].click();
	} }); 
	element.appendChild(button);
}
if(url[url.length-2] === "curators") {
	var element = document.getElementById("curator-action-bar");
	var button = document.createElement('div');
	button.classList.add('button');
	button.classList.add('grey');
	button.classList.add('small');
	button.textContent = " Add all followers ";
        document.getElementById("open").click();
	button.addEventListener("click", function(){ 
	if(document.getElementById("explore-header-closed").getAttribute("data-control") === "open") {document.getElementById("open").click();}
	for(var i = 0;i<document.getElementsByClassName("add-label").length;i++) {
			document.getElementsByClassName("add-label")[i].click();
	} }); 
	element.appendChild(button);
}  }, 500);
```
This could be implemented better as part of the studio-tools addon

**Reply by WorldLanguages** [reply]
So the 2 features are "delete all projects" and "add all followers"? 🤔

**Reply by GrahamSH-LLK** [reply]
> So the 2 features are "delete all projects" and "add all followers"? 🤔

Yeah, but the add all followers implementation looks sketchy (ie. I don't think it works?), and the projects one wouldn't work with pagination

**Reply by starkodder** [reply]
Yes i will add more soon.

**Reply by starkodder** [reply]
> 
> 
> > So the 2 features are "delete all projects" and "add all followers"? 🤔
> 
> Yeah, but the add all followers implementation looks sketchy (ie. I don't think it works?), and the projects one wouldn't work with pagination

I think 'add all followers' will work if it fetches a list of followers.

**Reply by GrahamSH-LLK** [reply]
> > > So the 2 features are "delete all projects" and "add all followers"? 🤔
> > 
> > 
> > Yeah, but the add all followers implementation looks sketchy (ie. I don't think it works?), and the projects one wouldn't work with pagination
> 
> I think 'add all followers' will work if it fetches a list of followers.

I forgot to mention: it's rejected

**Comment by starkodder** [comment]
I made a new code to promote people because scratch dosent always load every page.
Here it is (I will add button later)
`function getCookie(cname) {
  var name = cname + "=";
  var decodedCookie = decodeURIComponent(document.cookie);
  var ca = decodedCookie.split(';');
  for(var i = 0; i <ca.length; i++) {
    var c = ca[i];
    while (c.charAt(0) == ' ') {
      c = c.substring(1);
    }
    if (c.indexOf(name) == 0) {
      return c.substring(name.length, c.length);
    }
  }
  return "";
}

await fetch("https://scratch.mit.edu/site-api/users/curators-in/" + STUDIO_ID + "/promote/?usernames=" + USERNAME, {
    "credentials": "include",
    "headers": {
        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:88.0) Gecko/20100101 Firefox/88.0",
        "Accept": "*/*",
        "Accept-Language": "sv-SE,sv;q=0.8,en-US;q=0.5,en;q=0.3",
        "X-CSRFToken": getCookie('scratchcsrftoken'),
        "X-Requested-With": "XMLHttpRequest",
        "X-KL-Ajax-Request": "Ajax_Request"
    },
    "referrer": "https://scratch.mit.edu/studios/" + STUDIO_ID + "/curators/",
    "method": "PUT",
    "mode": "cors"
});`

**Comment by starkodder** [comment]
API links:
https://scratch.mit.edu/site-api/users/curators-in/STUDIO_ID/invite_curator/?usernames=USERNAME
https://scratch.mit.edu/site-api/users/curators-in/STUDIO_ID/promote/?usernames=USERNAME
https://scratch.mit.edu/site-api/users/curators-in/STUDIO_ID/remove/?usernames=USERNAME

**Comment by cobaltt7** [comment]
~Delete~ remove all projects: might be useful, you can make a PR to update studio-tools. but use the api
Promote curator: we already have
Add all followers: there's api limits, you can't use it if you have lots of followers

**Reply by WorldLanguages** [reply]
Not sure if delete all projects is fine. There's probably also api limits for that...

**Reply by cobaltt7** [reply]
o right
